pkg/guestagent/api/client/client.go

```diff
@@ -20,7 +18,7 @@ func NewGuestAgentClient(dialFn func(ctx context.Context) (net.Conn, error)) (*G
 		grpc.WithContextDialer(func(ctx context.Context, target string) (net.Conn, error) {
 			return dialFn(ctx)
 		}),
-		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithTransportCredentials(NewCredentials()),
```

Comment:
Could you explain the difference?

Reply:
Its more of a try out from this comment
https://github.com/open-telemetry/opentelemetry-helm-charts/issues/646#issuecomment-1832391092

I suspect, GRPC do something based on handshake response `SecurityLevel`. With insecure it will be NoSecurity. In this case it will be PrivacyAndIntegrity